### PR TITLE
exiv2: 0.27.3 -> 0.27.4

### PR DESCRIPTION
**Pull request by paperdigits**
###### Motivation for this change
Update exiv2 patches all known security issues and adds support for reading Canon CR3 files.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/exiv2/default.nix [hunk 1]
```diff
@@ -15,38 +15,17 @@
...
     owner = "exiv2";
     repo  = "exiv2";
     rev = "v${version}";
-    sha256 = "0d294yhcdw8ziybyd4rp5hzwknzik2sm0cz60ff7fljacv75bjpy";
+    sha256 = "0m1x79q6i5fw3gr9k0dw0bbl7ym27g9vbmxiamks6yw028xqwc5a";
   };
 
-  patches = [
-    # Fix aarch64 build https://github.com/Exiv2/exiv2/pull/1271
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/exiv2/default.nix:31]: Please remove the unused fetchpatch from inputs.

Comment (paperdigits) [reply pkgs/development/libraries/exiv2/default.nix:31]: Done, thanks.

**Comment by sikmir** [comment]
Broken on darwin.

**Comment by mweinelt** [comment]
> Broken on darwin.

Needs more information.

**Comment by sikmir** [comment]
> > Broken on darwin.
> 
> Needs more information.

```
[ERROR] The output of the testcase mismatch the reference
[INFO] The output has been saved to file /private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-exiv2-0.27.4.drv-0/source/test/tmp/conversions.out
[INFO] simply_diff:
/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-exiv2-0.27.4.drv-0/source/test/data/conversions.out: 204 lines
/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-exiv2-0.27.4.drv-0/source/test/tmp/conversions.out: 214 lines
The first mismatch is in line 46:
< Xmp.exif.UserComment                         LangAlt     1  lang="x-default" This is a JIS encoded Exif user comment. Or was it?
> Warning: Charset conversion required but no character mapping functionality available.

----------------------------------------------------------------------
Ran 24 tests in 37.348s

FAILED (errors=1)
make[3]: *** [CMakeFiles/tests.dir/build.make:76: CMakeFiles/tests] Error 1
```

**Pull request merged**